src/main/java/prefs/Preferences.java

```diff
@@ -18,13 +17,8 @@
 
     public GlobalConfig global;
 
-    public Preferences(boolean isTestMode) {
-        if (isTestMode) {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, TEST_CONFIG_FILE);
-        } else {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, GLOBAL_CONFIG_FILE);
-        }
-
+    public Preferences(Optional<String> configFileName) {
```

Comment:
Thinking about it again, can we flip this? When configFileName is empty, it changes `TEST_CONFIG_FILE`. We will supply `GLOBAL_CONFIG_FILE` for real application. My concern is that if we have a bug on creating preferences in tests, it may mess up actual preference of the developer. Defaulting to test config would be safer.

What do you think?